src/api-binder.coffee

```diff
@@ -338,6 +338,9 @@ module.exports = class APIBinder
 			(currentState, targetConfig, defaultConfig, deviceId) =>
 				currentConfig = currentState.local.config
 				Promise.mapSeries _.toPairs(currentConfig), ([ key, value ]) =>
+					# We want to disable local mode when joining a cloud
+					if key == 'SUPERVISOR_LOCAL_MODE'
+						return @config.set({ localMode: false })
```

Comment:
Feels kinda awkward to directly change the config here... If instead you just set `value = 'false'` here, the device state engine will take care of setting the config to false, in the same way as the VPN control var below.

But won't block on this since this will work fine too.